tests/upgrade/rules/test_no_additional_args_operators.py

```diff
@@ -0,0 +1,49 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from tempfile import NamedTemporaryFile
+from unittest import TestCase
+import textwrap
+from airflow.upgrade.rules.no_additional_args_in_operators import NoAdditionalArgsInOperatorsRule
+from tests.test_utils.db import clear_db_connections
+
+
+class TestNoAdditionalArgsInOperatorsRule(TestCase):
+    def tearDown(self):
+        clear_db_connections()
+
+    def test_check(self):
+        rule = NoAdditionalArgsInOperatorsRule()
+
+        assert isinstance(rule.title, str)
+        assert isinstance(rule.description, str)
+
+        with NamedTemporaryFile(mode='w', suffix='.py') as dag_file:
+            dag_file.write(
+                textwrap.dedent('''
+            from airflow import DAG
+            from airflow.utils.dates import days_ago
+            from airflow.operators.bash_operator import BashOperator
+
+            with DAG(dag_id="test", start_date=days_ago(0)):
+                BashOperator(task_id='test', bash_command="true", extra_param=42)
+                '''
+                                ))
+            dag_file.flush()
+            msgs = rule.check(dags_folder=dag_file.name)
```

Comment:
```suggestion
            msgs = rule.check()
```
If we have this argument only for test then we should remove it and use mock similar to this:
https://github.com/apache/airflow/blob/847820f91207dcee815ecda96e48cf62b0977580/tests/upgrade/rules/test_import_changes.py#L49-L61

Reply:
Noted.

Reply:
Ah, the Dependency Injection vs Mocking debate. :)

Reply:
Having slept on it, I personally find the DI approach easier to understand than mocking.

Reply:
@ashb personally I don't like features in code that are not used when running the code - it creates confusion when not commented

Reply:
Also true.

Reply:
@turbaszek Since you and I don't agree, and both approaches already exist in the code base, are you happy to let @Dr-Denzy decide on which approach to user?

Reply:
> @turbaszek Since you and I don't agree, and both approaches already exist in the code base, are you happy to let @Dr-Denzy decide on which approach to user?

Yes, no objection to keep it as it is - we already use this approach 👌